Lib/test/test_monitoring.py

```diff
@@ -1709,6 +1709,24 @@ def func(v=1):
             ('branch right', 'func', 6, 8),
             ('branch right', 'func', 2, 10)])
 
+    def test_callback_set_frame_lineno(self):
+        def func(s: str) -> int:
+            if s.startswith("t"):
+                return 1
+            else:
+                return 0
+
+        def callback(code, from_, to):
+            frame = sys._getframe(0)
```

Comment:
```suggestion
            frame = sys._getframe(1)
```

`sys._getframe(0)` is the current frame, which is not safe to modify.
We should probably raise an exception if we try to set the line number of the currently executing frame.